File: py/problem_9.py
#!/usr/bin/env python

from math import prod


def find_right_triangle():
    for a in range(1, 333):
        for b in range(a, 500):
            c = 1000 - a - b
            if a * a + b * b == c * c:
                return (a, b, c)


triangle = find_right_triangle()
print(triangle)
print(prod(triangle))

File: dart/problem_9.dart
#!/usr/bin/env dart

import 'iterables.dart';

List<int> find_right_triangle() {
  return range(1, 333)
      .expand((a) => range(a, 500).map((b) => [a, b, 1000 - a - b]))
      .where((t) => t[0] * t[0] + t[1] * t[1] == t[2] * t[2])
      .first;
}

void main() {
  var triangle = find_right_triangle();
  print(triangle);
  print(triangle.product());
}
